File: src/main/java/org/soraworld/violet/command/CommandViolet.java
package org.soraworld.violet.command;

import org.soraworld.violet.config.VioletManager;
import org.soraworld.violet.constant.Violets;
import rikka.api.command.ICommandSender;

import java.util.ArrayList;

public class CommandViolet extends IICommand {

    public CommandViolet(String perm, boolean onlyPlayer, VioletManager manager, String... aliases) {
        super(perm, onlyPlayer, manager, aliases);
        addSub(new IICommand(manager.adminPerm, false, manager, "lang") {
            public boolean execute(ICommandSender sender, ArrayList<String> args) {
                if (args.isEmpty()) {
                    manager.vSendKey(sender, Violets.KEY_GET_LANG, manager.lang);
                } else {
                    manager.lang = args.get(0);
                    manager.save();
                    manager.vSendKey(sender, Violets.KEY_SET_LANG, manager.lang);
                }
                return true;
            }
        });
        addSub(new IICommand(manager.adminPerm, false, manager, "save") {
            public boolean execute(ICommandSender sender, ArrayList<String> args) {
                if (manager.save()) {
                    manager.vSendKey(sender, Violets.KEY_CFG_SAVE);
                } else {
                    manager.vSendKey(sender, Violets.KEY_CFG_SAVE_FAIL);
                }
                return true;
            }
        });
        addSub(new IICommand(manager.adminPerm, false, manager, "debug") {
            public boolean execute(ICommandSender sender, ArrayList<String> args) {
                if (manager.debug) {
                    manager.debug = false;
                    manager.vSendKey(sender, Violets.KEY_DEBUG_OFF);
                } else {
                    manager.debug = true;
                    manager.vSendKey(sender, Violets.KEY_DEBUG_ON);
                }
                return true;
            }
        });
        addSub(new IICommand(manager.adminPerm, false, manager, "reload") {
            public boolean execute(ICommandSender sender, ArrayList<String> args) {
                if (manager.load()) {
                    manager.vSendKey(sender, Violets.KEY_CFG_LOAD);
                } else {
                    manager.vSendKey(sender, Violets.KEY_CFG_LOAD_FAIL);
                }
                return true;
            }
        });
    }

}

File: src/main/kotlin/org/soraworld/violet/command/CommandViolet.kt
package org.soraworld.violet.command

import org.bukkit.command.CommandSender
import org.soraworld.violet.config.IIConfig
import org.soraworld.violet.constant.Violets

open class CommandViolet(name: String, perm: String?, config: IIConfig) : IICommand(name, perm, config) {

    init {
        addSub(object : IICommand("lang", config.adminPerm, config) {
            override fun execute(sender: CommandSender, args: MutableList<String>): Boolean {
                if (args.isEmpty()) {
                    config.sendV(sender, Violets.KEY_GET_LANG, config.lang)
                } else {
                    config.lang = args[0]
                    config.save()
                    config.sendV(sender, Violets.KEY_SET_LANG, config.lang)
                }
                return true
            }
        })
        addSub(object : IICommand("save", config.adminPerm, config) {
            override fun execute(sender: CommandSender, args: MutableList<String>): Boolean {
                if (config.save()) {
                    config.sendV(sender, Violets.KEY_CFG_SAVE)
                } else {
                    config.sendV(sender, Violets.KEY_CFG_SAVE_FAIL)
                }
                return true
            }
        })
        addSub(object : IICommand("debug", config.adminPerm, config) {
            override fun execute(sender: CommandSender, args: MutableList<String>): Boolean {
                if (config.debug) {
                    config.debug = false
                    config.sendV(sender, Violets.KEY_DEBUG_OFF)
                } else {
                    config.debug = true
                    config.sendV(sender, Violets.KEY_DEBUG_ON)
                }
                return true
            }
        })
        addSub(object : IICommand("reload", config.adminPerm, config) {
            override fun execute(sender: CommandSender, args: MutableList<String>): Boolean {
                if (config.load()) {
                    config.sendV(sender, Violets.KEY_CFG_LOAD)
                } else {
                    config.sendV(sender, Violets.KEY_CFG_LOAD_FAIL)
                }
                return true
            }
        })
    }

}
